### PR TITLE
grpc create methods should return the created object (CAS-88)

### PR DESCRIPTION
**Pull request by jkryl**
This simplifies the code in the control plane and improves atomicity of the create methods (either everything succeeds or fails).

### CODE REVIEW

File: mayastor/src/grpc/mayastor_grpc.rs [hunk 1]
```diff
@@ -278,16 +204,31 @@ impl mayastor_server::Mayastor for MayastorSvc {
...
+    ) -> GrpcResult<Child> {
         let args = request.into_inner();
         trace!("{:?}", args);
         let uuid = args.uuid.clone();
         debug!("Adding child {} to nexus {} ...", args.uri, uuid);
-        locally! { async move {
-            nexus_lookup(&args.uuid)?.add_child(&args.uri, args.norebuild).await.map(|_| ())
+        // XXX How to avoid following spaghetti code if we cannot
+        // use '?' inside 'locally' macro?
+        let child = locally! { async move {
```
Comment (tiagolobocastro) [inline mayastor/src/grpc/mayastor_grpc.rs:214]: What if you put what's inside the locally into a function and then you just call it, would that work?

File: mayastor-test/test_nexus.js [hunk 2]
```diff
@@ -385,11 +386,44 @@ describe('nexus', function () {
...
+          `iscsi://${externIp}:${iscsiReplicaPort}/iqn.2019-05.io.openebs:disk1`
+        );
+        assert.equal(nexus.children[2].state, 'CHILD_ONLINE');
+      }
+
+      if (doUring()) {
+        const uringIndex = 3 + doIscsiReplica;
+        assert.equal(
+          nexus.children[uringIndex].uri,
+          `uring:///${uringFile}?blk_size=4096`
```
Comment (tiagolobocastro) [inline mayastor-test/test_nexus.js:417]: extra slash here

File: mayastor/src/grpc/mayastor_grpc.rs (hunk 1 above)
Comment (jkryl) [reply mayastor/src/grpc/mayastor_grpc.rs:214]: yes, I could do that. I can create a special function in nexus_grpc.rs and call it from here. Not the best solution but likely better than this code.

File: mayastor-test/test_nexus.js (hunk 2 above)
Comment (jkryl) [reply mayastor-test/test_nexus.js]: thanks. fixed

**Pull request merged**